clippy_lints/src/map_nil_fn.rs

```diff
@@ -0,0 +1,187 @@
+use rustc::hir;
+use rustc::lint::*;
+use rustc::ty;
+use syntax::codemap::Span;
+use utils::{in_macro, iter_input_pats, match_type, method_chain_args, snippet, span_lint_and_then};
+use utils::paths;
+
+#[derive(Clone)]
+pub struct Pass;
+
+/// **What it does:** Checks for usage of `Option.map(f)` where f is a nil
```

Comment:
What if we instead lint for `Option.map(f)` where the result isn't used? Covers non-unit functions too. The code can be similar to the must_use lint from rustc.
Lint is fine as is too though.

https://github.com/rust-lang/rust/blob/master/src/librustc_lint/unused.rs#L110